src/services/FeedProcessor/CoingeckoMultiProcessor.ts

```diff
@@ -1,18 +1,52 @@
 import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
 
 import {InputParams, OutputValues} from '../fetchers/CoingeckoPriceMultiFetcher.js';
-
-import {FeedFetcher} from '../../types/Feed.js';
+import {PriceDataRepository, PriceDataPayload, PriceValueType} from '../../repositories/PriceDataRepository.js';
 import {CoingeckoPriceMultiFetcher} from '../fetchers/index.js';
+import {FeedFetcher} from '../../types/Feed.js';
+import {FetcherName} from '../../types/fetchers.js';
+import TimeService from '../TimeService.js';
+import FeedChecker from '../FeedChecker.js';
 
 @injectable()
 export default class CoingeckoMultiProcessor {
   @inject(CoingeckoPriceMultiFetcher) coingeckoPriceMultiFetcher!: CoingeckoPriceMultiFetcher;
+  @inject(PriceDataRepository) private priceDataRepository!: PriceDataRepository;
+  @inject(TimeService) private timeService!: TimeService;
+  @inject(FeedChecker) private feedChecker!: FeedChecker;
+  @inject('Logger') private logger!: Logger;
+
+  static fetcherSource = '';
 
   async apply(feedFetchers: FeedFetcher[]): Promise<(number | undefined)[]> {
     const params = this.createParams(feedFetchers);
     const outputs = await this.coingeckoPriceMultiFetcher.apply(params);
 
+    const payloads: PriceDataPayload[] = [];
+
+    for (const [ix, output] of outputs.entries()) {
+      const result = this.feedChecker.getBaseAndQuote(feedFetchers[ix].symbol);
```

Comment:
`if (!output) continue;`

maybe this is more clear?

Reply:
same in other placec

Reply:
done